execution_engine/src/core/tracking_copy/mod.rs

```diff
@@ -402,6 +419,7 @@ impl<R: StateReader<Key, StoredValue>> TrackingCopy<R> {
     pub fn query(
         &self,
         correlation_id: CorrelationId,
+        config: &EngineConfig,
```

Comment:
Personally, I wouldn't pass in the whole `EngineConfig` – only the bits that are required by the method.